Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageGalleryView.swift

```diff
@@ -123,10 +127,19 @@ open class _ChatMessageGalleryView<ExtraData: ExtraDataTypes>: _View, ThemeProvi
             .flatMap(\.subviews)
             .forEach { $0.removeFromSuperview() }
         
+        // Prevents layout issue. Can be removed when CIS-981 is fixed.
+        guard !content.isEmpty else {
+            previewsContainerView.isHidden = true
+            return
+        }
+        
         // Add previews to the spots
-        for (preview, spot) in zip(content, itemSpots) {
-            spot.addSubview(preview)
-            preview.pin(to: spot)
+        if content.count == 3 {
+            itemSpots[0].embed(content[0])
+            itemSpots[1].embed(content[1])
+            itemSpots[3].embed(content[2])
+        } else {
+            zip(itemSpots, content).forEach { $0.embed($1) }
         }
```

Comment:
this is also because of CIS-981?

Reply:
No, this is because of Figma. In the three image layout, the first image should be the tall one, but in our case, it was the second image. This is laying them out properly :)